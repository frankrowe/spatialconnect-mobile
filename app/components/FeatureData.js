'use strict';
import React, { Component, PropTypes } from 'react';
import {
  ScrollView,
} from 'react-native';
import { find } from 'lodash';
import { Actions } from 'react-native-router-flux';
import Property from './Property';
import { propertyListStyles } from '../style/style';

class FeatureData extends Component {
  //returns true if feature belongs to gpkg store
  isEditable(stores, feature) {
    if (!feature.metadata) return false;
    if (!feature.metadata.storeId) return false;
    let store = find(stores,
       s => s.storeId == feature.metadata.storeId
    );
    return store.type === 'gpkg';
  }
  //hide Edit button if feature is not editable
  componentDidMount() {
    if (this.isEditable(this.props.stores, this.props.feature)) {
      Actions.refresh({rightTitle: 'Edit'});
    } else {
      Actions.refresh({rightTitle: ''});
    }
  }
  renderMetadata() {
    let feature = this.props.feature;
    let metadata = [];
    metadata.push({ name: 'ID', value: feature.id });
    if (feature.metadata) {
      for (let key in feature.metadata) {
        metadata.push({ name: key, value: feature.metadata[key]});
      }
    }
    return <Property name={'Metadata'} values={metadata} />;
  }
  renderProperties() {
    let feature = this.props.feature;
    if (Object.keys(feature.properties).length) {
      let fields = [];
      for (let key in feature.properties) {
        fields.push({ name: key, value: feature.properties[key] });
      }
      return <Property name={'Properties'} values={fields} />;
    } else return null;
  }
  renderLocation() {
    let feature = this.props.feature;
    return (feature.geometry && feature.geometry.type === 'Point') ?
      <Property name={'Location'} values={[
        {name: 'Latitude', value: feature.geometry.coordinates[1]},
        {name: 'Longitude', value: feature.geometry.coordinates[0]}]} />
      : null;
  }
  render() {
    return (
      <ScrollView style={propertyListStyles.container}>
        {this.renderMetadata()}
        {this.renderLocation()}
        {this.renderProperties()}
      </ScrollView>
    );
  }
}

FeatureData.propTypes = {
  feature: PropTypes.object.isRequired,
  stores: PropTypes.array.isRequired,
};

export default FeatureData;