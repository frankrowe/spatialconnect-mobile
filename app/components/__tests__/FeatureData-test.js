import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import feature from 'feature';
import FeatureData from '../FeatureData';

describe('<FeatureData />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <FeatureData feature={feature} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
