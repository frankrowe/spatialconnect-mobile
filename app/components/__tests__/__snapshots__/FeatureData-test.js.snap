exports[`<FeatureData /> renders correctly 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "marginTop": -1
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "column"
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#f0f1f4",
          "borderBottomWidth": 0,
          "borderColor": "#dddddd",
          "borderTopWidth": 0,
          "justifyContent": "center",
          "padding": 10
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold"
          }
        }>
        Metadata
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10
        }
      }>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold"
              }
            }>
            ID
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12
              }
            }>
            (null).REVGQVVMVF9TVE9SRS1CRDhDOUE5OC1DODg5LTQyNUYtQkY5Ni1EQ0IxNzE2NEM4Q0M=.NQ==
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold"
              }
            }>
            created_at
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12
              }
            }>
            2016-06-17 14:30:52 EDT
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold"
              }
            }>
            client
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12
              }
            }>
            BD8C9A98-C889-425F-BF96-DCB17164C8CC
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "column"
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#f0f1f4",
          "borderBottomWidth": 0,
          "borderColor": "#dddddd",
          "borderTopWidth": 0,
          "justifyContent": "center",
          "padding": 10
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold"
          }
        }>
        Location
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10
        }
      }>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold"
              }
            }>
            Latitude
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12
              }
            }>
            38.35388793147947
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold"
              }
            }>
            Longitude
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12
              }
            }>
            -75.61079864397236
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "column"
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#f0f1f4",
          "borderBottomWidth": 0,
          "borderColor": "#dddddd",
          "borderTopWidth": 0,
          "justifyContent": "center",
          "padding": 10
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold"
          }
        }>
        Properties
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10
        }
      }>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold"
              }
            }>
            team
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12
              }
            }>
            O\'s
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold"
              }
            }>
            why
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12
              }
            }>
            testing
          </Text>
        </Text>
      </View>
    </View>
  </View>
</ScrollView>
`;
